src/modules/deposit/service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { Deposit } from "../scraper/model/deposit.entity";
+
+@Injectable()
+export class DepositService {
+  constructor(@InjectRepository(Deposit) private depositRepository: Repository<Deposit>) {}
+
+  public async getDeposits(status = null, limit = 10, offset = 0) {
+    const where = status ? { status } : undefined;
+
+    const [deposits, total] = await Promise.all([
+      this.depositRepository.find({
+        where,
+        take: limit,
+        skip: offset,
+        order: {
+          depositDate: "desc",
+        },
+      }),
+      this.depositRepository.count({ where }),
+    ]);
+
+    return {
+      deposits,
```

Comment:
I think we need to select only a subset of columns and some of them needs to be renamed/modified in order to match the response expected by the FE

depositDate (from db) -> depositTime (UNIX timestamp)
fillTxs -> the FE is expecting an array of strings, but the DB column is an array of objects.

Reply:
Yea, so currently I am handling the formatting on the client. See https://github.com/across-protocol/frontend-v2/pull/221/files#diff-bbece6ae0b2388ae96ac90b7bdf7db01dee1e8f61e4ff3a008d1cf002b83abc2R36-R49.

I mean these additional columns could be handy at some point? Or is the goal of this API to mirror exactly the structure of the on-chain `Transfers`?

Reply:
> I mean these additional columns could be handy at some point?

This is a good question. At this moment we don't have plans to use the additional columns in any way.